src/main/java/org/jboss/japit/analyser/JarAnalyser.java

```diff
@@ -30,95 +30,101 @@
-                            classArchive.setOriginalJavaFile(ctClz.getClassFile().getSourceFile());
-                            classArchive.setDeclaredMethodsCount(ctClz.getDeclaredMethods().length);
-                            classArchive.setDeclaredFieldsCount(ctClz.getDeclaredFields().length);
-                            
-                            TreeSet<String> methods = new TreeSet<String>();
-                            for (CtMethod method : ctClz.getMethods()) {
-                                methods.add(Modifier.toString(method.getModifiers()) + " " + getReturnTypeName(method)
-                                        + " " + method.getLongName());
-                            }
-                            classArchive.setMethods(methods);
+            try (JarFile jar = new JarFile(file)) {
+                JarArchive jarArchive = new JarArchive(file.getName(), file.getCanonicalPath());
+                TreeSet<ClassDetails> classes = new TreeSet<ClassDetails>();
 
-                            TreeSet<String> fields = new TreeSet<String>();
-                            for (CtField field : ctClz.getFields()) {
-                                fields.add(Modifier.toString(field.getModifiers()) + " " + field.getFieldInfo().toString());
-                            }
-                            classArchive.setFields(fields);
+                Enumeration<JarEntry> jarEntries = jar.entries();
+                JAR_ELEMENT: while (jarEntries.hasMoreElements()) {
+                    JarEntry jarEntry = jarEntries.nextElement();
+                    String entryName = jarEntry.getName();
+                    if (entryName.endsWith(".class")) {
+                        try (InputStream entryStream = jar.getInputStream(jarEntry)) {
 
-                            classes.add(classArchive);
-                        }
-                    } catch (Throwable ie) {
-                        System.err.println("Exception thrown when processing " + entryName);
-                        ie.printStackTrace(System.err);
-                    } finally {
-                        if (entryStream != null) {
-                            try {
-                                entryStream.close();
-                            } catch (IOException ex) {
-                                System.err.println("EntryStream couldn't be closed for " + entryName);
-                                ex.printStackTrace(System.err);
+                            ClassPool classPool = new ClassPool(true);
+                            CtClass ctClz = classPool.makeClass(entryStream);
+                            String className = ctClz.getName();
+                            if (selectedFQCN == null || className.equals(selectedFQCN)) {
+
+                                for (Pattern skipPattern : skipPatterns) {
+                                    if (skipPattern.matcher(className).matches()) {
+                                        continue JAR_ELEMENT;
```

Comment:
Labeled continue statement is not nice

I'd rather see new `processEntry` method (using code from line 73 - 117) returning Optional<ClassDetails>
classes.add would be invoked after Optional.ifPresent check

Reply:
IMO the labeled `continue` is much nicer and safer than a simple one. Nevertheless, I can do more Java 8 stuff here ;)

Reply:
Thanks, will check it later today or tomorrow.

I see you have fun with hazelcast/* repositories, enjoy the new challenges!